src/Angor/Shared/Services/NostrCommunicationFactory.cs

```diff
@@ -117,7 +117,7 @@ public bool EoseEventReceivedOnAllRelays(string subscription)
     public void MonitoringEoseReceivedOnSubscription(string subscription)
     {
         _logger.LogInformation($"Started monitoring subscription {subscription}");
-        _eoseCalledOnSubscriptionClients.Add(subscription, new List<string>());
+        _eoseCalledOnSubscriptionClients.TryAdd(subscription, new List<string>());
```

Comment:
Better to add a boolean returned from the method and check it in the caller so we can identify where the duplication is from

Reply:
So you mean a `return TryAdd(...` ?

Reply:
Instead I logged a warning is this good enough?
this was called from `RelaySubscriptionsHandling.TryAddEoseAction`

Reply:
Return the boolean and we can handle it from the caller

Reply:
Handle how? log it at the caller instead?

Reply:
Yes, that way we can tell where the duplication happens and why